llvm/include/llvm/Analysis/MemorySSAUpdater.h

```diff
@@ -190,7 +190,8 @@ class MemorySSAUpdater {
   /// inaccessible and it *must* have removeMemoryAccess called on it.
   MemoryAccess *createMemoryAccessInBB(Instruction *I, MemoryAccess *Definition,
                                        const BasicBlock *BB,
-                                       MemorySSA::InsertionPlace Point);
+                                       MemorySSA::InsertionPlace Point,
+                                       bool CreationMustSucceed = true);
```

Comment:
This is an ABI-breaking change. Instead of an optional argument, you need to add two functions and forward one to the other.